src/main/java/org/opensearch/neuralsearch/search/collector/HybridTopFieldDocSortCollector.java

```diff
@@ -254,6 +259,9 @@ protected void initializePriorityQueuesWithComparators(LeafReaderContext context
                     initializeLeafFieldComparators(context, i);
                 }
             }
+            if (Objects.isNull(fieldValueLeafTrackers)) {
```

Comment:
QQ: Do you want this code to be executed 
 a. Once per shard?
 b. Once per segment?
 c. Once per entire search flow?

Reply:
I think this code can be shifted under above if(compoundScores==null) condition.